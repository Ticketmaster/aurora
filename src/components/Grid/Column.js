import PropTypes from "prop-types";
import styled from "styled-components";
import constants from "../../theme/constants";
import spacing from "../../theme/spacing";

const MAX_COLUMNS = 12;

const getSize = val => val / MAX_COLUMNS * 100;

const getPadding = size => props => {
  if ((props[size] || props.large || props.medium || props.small) === 12)
    return 0;

  return `${spacing.gutters[size] || spacing.gutters.mediumAndUp}px`;
};

const Column = styled.div.attrs({ "data-col": true })`
  float: left;
  padding-right: ${getPadding("small")};
  width: ${props => getSize(props.small)}%;

  @media screen and ${constants.breakpoints.mediumAndUp} {
    width: ${props => getSize(props.medium || props.small)}%;
    padding-right: ${getPadding("medium")};
  }

  @media screen and ${constants.breakpoints.largeAndUp} {
    width: ${props => getSize(props.large || props.medium || props.small)}%;
    padding-right: ${getPadding("large")};
  }

  @media screen and ${constants.breakpoints.xlAndUp} {
    width: ${props =>
      getSize(props.xLarge || props.large || props.medium || props.small)}%;
    padding-right: ${getPadding("xLarge")};
  }
`;

Column.propTypes = {
  small: PropTypes.number,
  medium: PropTypes.number,
  large: PropTypes.number,
  xLarge: PropTypes.number
};

Column.defaultProps = {
  small: 12,
  medium: null,
  large: null,
  xLarge: null
};

export default Column;
