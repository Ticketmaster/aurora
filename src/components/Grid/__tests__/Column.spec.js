import React from "react";
import renderer from "react-test-renderer";

import Column from "../Column";

describe("Column", () => {
  it("warns with invalid column size", () => {
    console.error = jest.fn(); // eslint-disable-line

    renderer.create(
      <Column small={-1}>
        <span>Content!</span>
      </Column>
    );

    expect(console.error).toHaveBeenCalled(); // eslint-disable-line
  });
});
