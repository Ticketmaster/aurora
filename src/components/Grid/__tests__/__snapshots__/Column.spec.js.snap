// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column renders correctly when small value is 0 1`] = `
<div
  class="sc-bdVaJa jCYtFM"
>
  <span>
    Content!
  </span>
</div>
`;

exports[`Column renders correctly when value is 0 1`] = `
<div
  class="sc-bdVaJa fFxxKd"
>
  <span>
    Content!
  </span>
</div>
`;

exports[`Column renders correctly without any props 1`] = `
<div
  class="sc-bdVaJa fVSYzJ"
>
  <span>
    Content!
  </span>
</div>
`;
